{
    "nbformat_minor": 1, 
    "cells": [
        {
            "source": "Shape data in \"77_cancer_proteomes_CPTAC_itraq.csv\" from colum mode to raw mode.\n\n\nThe data are the raw exported in CSV format and loaded in a DashDB cloud instance (jdbc:db2://dashdb-entry-yp-dal09-10.services.dal.bluemix.net:50000/BLUDB:retrieveMessagesFromServerOnGetMessage=true)\nin  a table with following DDL:\n\nCREATE TABLE \"DASH13540\".\"BREASTCANCER_GENOMES\" (\n\t\t\"RefSeq_accession_number\" CHAR(100 OCTETS), \n\t\t\"gene_symbol\" CHAR(20 OCTETS), \n\t\t\"gene_name\" CHAR(250 OCTETS), \n\t\t\"Patient\" CHAR(28 OCTETS),\n\t\t\"Value\" DOUBLE, \n\t\t\"Gene\" CHAR(250 OCTETS) NOT NULL, \n\t\t\"Patient_ID\" CHAR(24 OCTETS) NOT NULL\n\t)\n\tORGANIZE BY COLUMN\n\tDATA CAPTURE NONE;\n\nWith following SQL i verify data duplication:\n\nselect distinct \"Patient_ID\"\n from \"DASH13540\".\"BREASTCANCER_GENOMES\"\n group by \"Gene\",\n          \"Patient_ID\"\n having count(*)>1; \n \nI observer that 3 Patient_ID (AO-A12D, C8-A131, A0-A12B) have duplicated column in term of proteomes values having a single status record in Patient data.\n\nConsquently i delete those column generating a new csv input file \"77_cancer_proteomes_CPTAC_itraq_NODouble.csv\" and again a reshaped it\n\nI also observed that last 3 column in data matrix refer to heatly patient (without record in patient statis data). So i change column in HEALTY1,2,3 and i add those 3 pseudo patient to patient record.\n\nAgain i create a new csv file \"clinical_data_breast_cancer_withHealty.csv\" and i use this\n\nIn notebook i maintain alla passage", 
            "cell_type": "markdown", 
            "metadata": {
                "collapsed": true
            }
        }, 
        {
            "execution_count": 15, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "output_type": "stream", 
                    "name": "stdout", 
                    "text": "Requirement already up-to-date: numpy in /home/dsxuser/.local/lib/python3.5/site-packages\nRequirement already up-to-date: scipy in /home/dsxuser/.local/lib/python3.5/site-packages\nRequirement already up-to-date: matplotlib in /home/dsxuser/.local/lib/python3.5/site-packages\nRequirement already up-to-date: ipython in /home/dsxuser/.local/lib/python3.5/site-packages\nRequirement already up-to-date: jupyter in /home/dsxuser/.local/lib/python3.5/site-packages\nRequirement already up-to-date: pandas in /home/dsxuser/.local/lib/python3.5/site-packages\nRequirement already up-to-date: sympy in /home/dsxuser/.local/lib/python3.5/site-packages\nRequirement not upgraded as not directly required: kiwisolver>=1.0.1 in /home/dsxuser/.local/lib/python3.5/site-packages (from matplotlib)\nRequirement not upgraded as not directly required: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from matplotlib)\nRequirement not upgraded as not directly required: pytz in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from matplotlib)\nRequirement not upgraded as not directly required: python-dateutil>=2.1 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from matplotlib)\nRequirement not upgraded as not directly required: cycler>=0.10 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from matplotlib)\nRequirement not upgraded as not directly required: six>=1.10 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from matplotlib)\nRequirement not upgraded as not directly required: simplegeneric>0.8 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: prompt-toolkit<2.0.0,>=1.0.15 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: pickleshare in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: traitlets>=4.2 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: pygments in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: pexpect; sys_platform != \"win32\" in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: backcall in /home/dsxuser/.local/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: decorator in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: setuptools>=18.5 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: jedi>=0.10 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipython)\nRequirement not upgraded as not directly required: ipywidgets in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from jupyter)\nRequirement not upgraded as not directly required: qtconsole in /home/dsxuser/.local/lib/python3.5/site-packages (from jupyter)\nRequirement not upgraded as not directly required: nbconvert in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from jupyter)\nRequirement not upgraded as not directly required: notebook in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from jupyter)\nRequirement not upgraded as not directly required: ipykernel in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from jupyter)\nRequirement not upgraded as not directly required: jupyter-console in /home/dsxuser/.local/lib/python3.5/site-packages (from jupyter)\nRequirement not upgraded as not directly required: mpmath>=0.19 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from sympy)\nRequirement not upgraded as not directly required: wcwidth in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from prompt-toolkit<2.0.0,>=1.0.15->ipython)\nRequirement not upgraded as not directly required: ipython_genutils in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from traitlets>=4.2->ipython)\nRequirement not upgraded as not directly required: nbformat>=4.2.0 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipywidgets->jupyter)\nRequirement not upgraded as not directly required: widgetsnbextension~=3.1.0 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipywidgets->jupyter)\nRequirement not upgraded as not directly required: jupyter-client>=4.1 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from qtconsole->jupyter)\nRequirement not upgraded as not directly required: jupyter-core in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from qtconsole->jupyter)\nRequirement not upgraded as not directly required: mistune>=0.7.4 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from nbconvert->jupyter)\nRequirement not upgraded as not directly required: jinja2 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from nbconvert->jupyter)\nRequirement not upgraded as not directly required: entrypoints>=0.2.2 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from nbconvert->jupyter)\nRequirement not upgraded as not directly required: bleach in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from nbconvert->jupyter)\nRequirement not upgraded as not directly required: pandocfilters>=1.4.1 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from nbconvert->jupyter)\nRequirement not upgraded as not directly required: testpath in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from nbconvert->jupyter)\nRequirement not upgraded as not directly required: tornado>=4.0 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from ipykernel->jupyter)\nRequirement not upgraded as not directly required: jsonschema!=2.5.0,>=2.4 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from nbformat>=4.2.0->ipywidgets->jupyter)\nRequirement not upgraded as not directly required: pyzmq>=13 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from jupyter-client>=4.1->qtconsole->jupyter)\nRequirement not upgraded as not directly required: MarkupSafe>=0.23 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from jinja2->nbconvert->jupyter)\nRequirement not upgraded as not directly required: html5lib>=0.99999999 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from bleach->nbconvert->jupyter)\nRequirement not upgraded as not directly required: webencodings in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from html5lib>=0.99999999->bleach->nbconvert->jupyter)\nRequirement already up-to-date: pixiedust in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages\nRequirement not upgraded as not directly required: markdown in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from pixiedust)\nRequirement not upgraded as not directly required: lxml in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from pixiedust)\nRequirement not upgraded as not directly required: mpld3 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from pixiedust)\nRequirement not upgraded as not directly required: geojson in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from pixiedust)\nRequirement not upgraded as not directly required: astunparse in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from pixiedust)\nRequirement not upgraded as not directly required: colour in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from pixiedust)\nRequirement not upgraded as not directly required: wheel<1.0,>=0.23.0 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from astunparse->pixiedust)\nRequirement not upgraded as not directly required: six<2.0,>=1.6.1 in /opt/conda/envs/DSX-Python35/lib/python3.5/site-packages (from astunparse->pixiedust)\n"
                }
            ], 
            "source": "! pip install --upgrade --user numpy scipy matplotlib ipython jupyter pandas sympy \n! pip install --user --upgrade pixiedust"
        }, 
        {
            "execution_count": 16, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [], 
            "source": "from io import StringIO\nfrom scipy import signal\nfrom matplotlib import pylab as pl\nfrom mpl_toolkits.mplot3d.axes3d import Axes3D\nimport requests\nimport json\nimport pandas as pd\nimport numpy as np\nimport matplotlib.pyplot as plt\nimport pixiedust\nimport sys\nimport types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\nfrom project_lib import Project"
        }, 
        {
            "execution_count": 17, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [], 
            "source": "#The project token is an authorization token that is used to access project resources like data sources, connections, and used by platform APIs.\nproject = Project(project_id='8cec06cd-ebbe-4f7a-9f66-8e1e40722f95', project_access_token='p-1298f2f1ff374652829046e40c4f21f2e59886f1')\npc = project.project_context"
        }, 
        {
            "execution_count": 18, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 18, 
                    "metadata": {}, 
                    "data": {
                        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>RefSeq_accession_number</th>\n      <th>gene_symbol</th>\n      <th>gene_name</th>\n      <th>AO-A12D.01TCGA</th>\n      <th>C8-A131.01TCGA</th>\n      <th>AO-A12B.01TCGA</th>\n      <th>BH-A18Q.02TCGA</th>\n      <th>C8-A130.02TCGA</th>\n      <th>C8-A138.03TCGA</th>\n      <th>E2-A154.03TCGA</th>\n      <th>...</th>\n      <th>AO-A12B.34TCGA</th>\n      <th>A2-A0SW.35TCGA</th>\n      <th>AO-A0JL.35TCGA</th>\n      <th>BH-A0BV.35TCGA</th>\n      <th>A2-A0YM.36TCGA</th>\n      <th>BH-A0C7.36TCGA</th>\n      <th>A2-A0SX.36TCGA</th>\n      <th>263d3f-I.CPTAC</th>\n      <th>blcdb9-I.CPTAC</th>\n      <th>c4155b-C.CPTAC</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NP_958782</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1</td>\n      <td>1.096131</td>\n      <td>2.609943</td>\n      <td>-0.659828</td>\n      <td>0.195341</td>\n      <td>-0.494060</td>\n      <td>2.765081</td>\n      <td>0.862659</td>\n      <td>...</td>\n      <td>-0.963904</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.065838</td>\n      <td>0.655850</td>\n      <td>-0.552212</td>\n      <td>-0.398560</td>\n      <td>0.598585</td>\n      <td>-0.191285</td>\n      <td>0.566975</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NP_958785</td>\n      <td>NaN</td>\n      <td>plectin isoform 1g</td>\n      <td>1.111370</td>\n      <td>2.650422</td>\n      <td>-0.648742</td>\n      <td>0.215413</td>\n      <td>-0.503899</td>\n      <td>2.779709</td>\n      <td>0.870186</td>\n      <td>...</td>\n      <td>-0.938210</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.055893</td>\n      <td>0.658143</td>\n      <td>-0.547749</td>\n      <td>-0.392601</td>\n      <td>0.606697</td>\n      <td>-0.183918</td>\n      <td>0.578702</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NP_958786</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1a</td>\n      <td>1.111370</td>\n      <td>2.650422</td>\n      <td>-0.654285</td>\n      <td>0.215413</td>\n      <td>-0.500619</td>\n      <td>2.779709</td>\n      <td>0.870186</td>\n      <td>...</td>\n      <td>-0.943919</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.065838</td>\n      <td>0.655850</td>\n      <td>-0.552212</td>\n      <td>-0.392601</td>\n      <td>0.603993</td>\n      <td>-0.186022</td>\n      <td>0.576747</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NP_000436</td>\n      <td>NaN</td>\n      <td>plectin isoform 1c</td>\n      <td>1.107561</td>\n      <td>2.646374</td>\n      <td>-0.632113</td>\n      <td>0.205377</td>\n      <td>-0.510459</td>\n      <td>2.797995</td>\n      <td>0.866423</td>\n      <td>...</td>\n      <td>-0.935355</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.055893</td>\n      <td>0.655850</td>\n      <td>-0.552212</td>\n      <td>-0.392601</td>\n      <td>0.603993</td>\n      <td>-0.186022</td>\n      <td>0.576747</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NP_958781</td>\n      <td>NaN</td>\n      <td>plectin isoform 1e</td>\n      <td>1.115180</td>\n      <td>2.646374</td>\n      <td>-0.640428</td>\n      <td>0.215413</td>\n      <td>-0.503899</td>\n      <td>2.787023</td>\n      <td>0.870186</td>\n      <td>...</td>\n      <td>-0.935355</td>\n      <td>-0.503853</td>\n      <td>-0.10668</td>\n      <td>-0.062523</td>\n      <td>0.651264</td>\n      <td>-0.556675</td>\n      <td>-0.395581</td>\n      <td>0.603993</td>\n      <td>-0.167079</td>\n      <td>0.576747</td>\n    </tr>\n  </tbody>\n</table>\n<p>5 rows \u00d7 86 columns</p>\n</div>", 
                        "text/plain": "  RefSeq_accession_number gene_symbol           gene_name  AO-A12D.01TCGA  \\\n0               NP_958782        PLEC   plectin isoform 1        1.096131   \n1               NP_958785         NaN  plectin isoform 1g        1.111370   \n2               NP_958786        PLEC  plectin isoform 1a        1.111370   \n3               NP_000436         NaN  plectin isoform 1c        1.107561   \n4               NP_958781         NaN  plectin isoform 1e        1.115180   \n\n   C8-A131.01TCGA  AO-A12B.01TCGA  BH-A18Q.02TCGA  C8-A130.02TCGA  \\\n0        2.609943       -0.659828        0.195341       -0.494060   \n1        2.650422       -0.648742        0.215413       -0.503899   \n2        2.650422       -0.654285        0.215413       -0.500619   \n3        2.646374       -0.632113        0.205377       -0.510459   \n4        2.646374       -0.640428        0.215413       -0.503899   \n\n   C8-A138.03TCGA  E2-A154.03TCGA       ...        AO-A12B.34TCGA  \\\n0        2.765081        0.862659       ...             -0.963904   \n1        2.779709        0.870186       ...             -0.938210   \n2        2.779709        0.870186       ...             -0.943919   \n3        2.797995        0.866423       ...             -0.935355   \n4        2.787023        0.870186       ...             -0.935355   \n\n   A2-A0SW.35TCGA  AO-A0JL.35TCGA  BH-A0BV.35TCGA  A2-A0YM.36TCGA  \\\n0       -0.487772        -0.10668       -0.065838        0.655850   \n1       -0.487772        -0.10668       -0.055893        0.658143   \n2       -0.487772        -0.10668       -0.065838        0.655850   \n3       -0.487772        -0.10668       -0.055893        0.655850   \n4       -0.503853        -0.10668       -0.062523        0.651264   \n\n   BH-A0C7.36TCGA  A2-A0SX.36TCGA  263d3f-I.CPTAC  blcdb9-I.CPTAC  \\\n0       -0.552212       -0.398560        0.598585       -0.191285   \n1       -0.547749       -0.392601        0.606697       -0.183918   \n2       -0.552212       -0.392601        0.603993       -0.186022   \n3       -0.552212       -0.392601        0.603993       -0.186022   \n4       -0.556675       -0.395581        0.603993       -0.167079   \n\n   c4155b-C.CPTAC  \n0        0.566975  \n1        0.578702  \n2        0.576747  \n3        0.576747  \n4        0.576747  \n\n[5 rows x 86 columns]"
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "# The code was removed by Watson Studio for sharing."
        }, 
        {
            "execution_count": 19, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 19, 
                    "metadata": {}, 
                    "data": {
                        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Complete TCGA ID</th>\n      <th>Gender</th>\n      <th>Age at Initial Pathologic Diagnosis</th>\n      <th>ER Status</th>\n      <th>PR Status</th>\n      <th>HER2 Final Status</th>\n      <th>Tumor</th>\n      <th>Tumor--T1 Coded</th>\n      <th>Node</th>\n      <th>Node-Coded</th>\n      <th>...</th>\n      <th>PAM50 mRNA</th>\n      <th>SigClust Unsupervised mRNA</th>\n      <th>SigClust Intrinsic mRNA</th>\n      <th>miRNA Clusters</th>\n      <th>methylation Clusters</th>\n      <th>RPPA Clusters</th>\n      <th>CN Clusters</th>\n      <th>Integrated Clusters (with PAM50)</th>\n      <th>Integrated Clusters (no exp)</th>\n      <th>Integrated Clusters (unsup exp)</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>TCGA-A2-A0T2</td>\n      <td>FEMALE</td>\n      <td>66</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>T3</td>\n      <td>T_Other</td>\n      <td>N3</td>\n      <td>Positive</td>\n      <td>...</td>\n      <td>Basal-like</td>\n      <td>0</td>\n      <td>-13</td>\n      <td>3</td>\n      <td>5</td>\n      <td>Basal</td>\n      <td>3</td>\n      <td>2</td>\n      <td>2</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>TCGA-A2-A0CM</td>\n      <td>FEMALE</td>\n      <td>40</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>T2</td>\n      <td>T_Other</td>\n      <td>N0</td>\n      <td>Negative</td>\n      <td>...</td>\n      <td>Basal-like</td>\n      <td>-12</td>\n      <td>-13</td>\n      <td>4</td>\n      <td>4</td>\n      <td>Basal</td>\n      <td>4</td>\n      <td>2</td>\n      <td>1</td>\n      <td>1</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>TCGA-BH-A18V</td>\n      <td>FEMALE</td>\n      <td>48</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>T2</td>\n      <td>T_Other</td>\n      <td>N1</td>\n      <td>Positive</td>\n      <td>...</td>\n      <td>Basal-like</td>\n      <td>-12</td>\n      <td>-13</td>\n      <td>5</td>\n      <td>5</td>\n      <td>Basal</td>\n      <td>1</td>\n      <td>2</td>\n      <td>2</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>TCGA-BH-A18Q</td>\n      <td>FEMALE</td>\n      <td>56</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>T2</td>\n      <td>T_Other</td>\n      <td>N1</td>\n      <td>Positive</td>\n      <td>...</td>\n      <td>Basal-like</td>\n      <td>-12</td>\n      <td>-13</td>\n      <td>5</td>\n      <td>5</td>\n      <td>Basal</td>\n      <td>1</td>\n      <td>2</td>\n      <td>2</td>\n      <td>2</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>TCGA-BH-A0E0</td>\n      <td>FEMALE</td>\n      <td>38</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>Negative</td>\n      <td>T3</td>\n      <td>T_Other</td>\n      <td>N3</td>\n      <td>Positive</td>\n      <td>...</td>\n      <td>Basal-like</td>\n      <td>0</td>\n      <td>-13</td>\n      <td>5</td>\n      <td>5</td>\n      <td>Basal</td>\n      <td>1</td>\n      <td>2</td>\n      <td>2</td>\n      <td>2</td>\n    </tr>\n  </tbody>\n</table>\n<p>5 rows \u00d7 30 columns</p>\n</div>", 
                        "text/plain": "  Complete TCGA ID  Gender  Age at Initial Pathologic Diagnosis ER Status  \\\n0     TCGA-A2-A0T2  FEMALE                                   66  Negative   \n1     TCGA-A2-A0CM  FEMALE                                   40  Negative   \n2     TCGA-BH-A18V  FEMALE                                   48  Negative   \n3     TCGA-BH-A18Q  FEMALE                                   56  Negative   \n4     TCGA-BH-A0E0  FEMALE                                   38  Negative   \n\n  PR Status HER2 Final Status Tumor Tumor--T1 Coded Node Node-Coded  \\\n0  Negative          Negative    T3         T_Other   N3   Positive   \n1  Negative          Negative    T2         T_Other   N0   Negative   \n2  Negative          Negative    T2         T_Other   N1   Positive   \n3  Negative          Negative    T2         T_Other   N1   Positive   \n4  Negative          Negative    T3         T_Other   N3   Positive   \n\n                ...                 PAM50 mRNA SigClust Unsupervised mRNA  \\\n0               ...                 Basal-like                          0   \n1               ...                 Basal-like                        -12   \n2               ...                 Basal-like                        -12   \n3               ...                 Basal-like                        -12   \n4               ...                 Basal-like                          0   \n\n  SigClust Intrinsic mRNA miRNA Clusters methylation Clusters RPPA Clusters  \\\n0                     -13              3                    5         Basal   \n1                     -13              4                    4         Basal   \n2                     -13              5                    5         Basal   \n3                     -13              5                    5         Basal   \n4                     -13              5                    5         Basal   \n\n   CN Clusters  Integrated Clusters (with PAM50)  \\\n0            3                                 2   \n1            4                                 2   \n2            1                                 2   \n3            1                                 2   \n4            1                                 2   \n\n   Integrated Clusters (no exp)  Integrated Clusters (unsup exp)  \n0                             2                                2  \n1                             1                                1  \n2                             2                                2  \n3                             2                                2  \n4                             2                                2  \n\n[5 rows x 30 columns]"
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "body = client_8a4fc8dc8bf54832ab62f63af0c03772.get_object(Bucket='breastcancerproteomes-donotdelete-pr-zmcmbnmwpo1wza',Key='clinical_data_breast_cancer.csv')['Body']\n# add missing __iter__ method, so pandas accepts body as file-like object\nif not hasattr(body, \"__iter__\"): body.__iter__ = types.MethodType( __iter__, body )\n\npatientData = pd.read_csv(body)\npatientData.head()"
        }, 
        {
            "execution_count": 20, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 20, 
                    "metadata": {}, 
                    "data": {
                        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>RefSeq_accession_number</th>\n      <th>gene_symbol</th>\n      <th>gene_name</th>\n      <th>Patient</th>\n      <th>Value</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NP_958782</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.096131</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NP_958785</td>\n      <td>NaN</td>\n      <td>plectin isoform 1g</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.111370</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NP_958786</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1a</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.111370</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NP_000436</td>\n      <td>NaN</td>\n      <td>plectin isoform 1c</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.107561</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NP_958781</td>\n      <td>NaN</td>\n      <td>plectin isoform 1e</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.115180</td>\n    </tr>\n  </tbody>\n</table>\n</div>", 
                        "text/plain": "  RefSeq_accession_number gene_symbol           gene_name         Patient  \\\n0               NP_958782        PLEC   plectin isoform 1  AO-A12D.01TCGA   \n1               NP_958785         NaN  plectin isoform 1g  AO-A12D.01TCGA   \n2               NP_958786        PLEC  plectin isoform 1a  AO-A12D.01TCGA   \n3               NP_000436         NaN  plectin isoform 1c  AO-A12D.01TCGA   \n4               NP_958781         NaN  plectin isoform 1e  AO-A12D.01TCGA   \n\n      Value  \n0  1.096131  \n1  1.111370  \n2  1.111370  \n3  1.107561  \n4  1.115180  "
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "reshapedData=pd.melt(orgData,id_vars=[\"RefSeq_accession_number\",\"gene_symbol\",\"gene_name\"],var_name=\"Patient\",value_name=\"Value\")\nreshapedData.head()"
        }, 
        {
            "execution_count": 21, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 21, 
                    "metadata": {}, 
                    "data": {
                        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>RefSeq_accession_number</th>\n      <th>gene_symbol</th>\n      <th>gene_name</th>\n      <th>Patient</th>\n      <th>Value</th>\n      <th>Gene</th>\n      <th>Patient_ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NP_958782</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.096131</td>\n      <td>NP_958782-PLEC (plectin isoform 1)</td>\n      <td>TCGA-AO-A12D</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NP_958785</td>\n      <td>NaN</td>\n      <td>plectin isoform 1g</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.111370</td>\n      <td>NP_958785-# (plectin isoform 1g)</td>\n      <td>TCGA-AO-A12D</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NP_958786</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1a</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.111370</td>\n      <td>NP_958786-PLEC (plectin isoform 1a)</td>\n      <td>TCGA-AO-A12D</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NP_000436</td>\n      <td>NaN</td>\n      <td>plectin isoform 1c</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.107561</td>\n      <td>NP_000436-# (plectin isoform 1c)</td>\n      <td>TCGA-AO-A12D</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NP_958781</td>\n      <td>NaN</td>\n      <td>plectin isoform 1e</td>\n      <td>AO-A12D.01TCGA</td>\n      <td>1.115180</td>\n      <td>NP_958781-# (plectin isoform 1e)</td>\n      <td>TCGA-AO-A12D</td>\n    </tr>\n  </tbody>\n</table>\n</div>", 
                        "text/plain": "  RefSeq_accession_number gene_symbol           gene_name         Patient  \\\n0               NP_958782        PLEC   plectin isoform 1  AO-A12D.01TCGA   \n1               NP_958785         NaN  plectin isoform 1g  AO-A12D.01TCGA   \n2               NP_958786        PLEC  plectin isoform 1a  AO-A12D.01TCGA   \n3               NP_000436         NaN  plectin isoform 1c  AO-A12D.01TCGA   \n4               NP_958781         NaN  plectin isoform 1e  AO-A12D.01TCGA   \n\n      Value                                 Gene    Patient_ID  \n0  1.096131   NP_958782-PLEC (plectin isoform 1)  TCGA-AO-A12D  \n1  1.111370     NP_958785-# (plectin isoform 1g)  TCGA-AO-A12D  \n2  1.111370  NP_958786-PLEC (plectin isoform 1a)  TCGA-AO-A12D  \n3  1.107561     NP_000436-# (plectin isoform 1c)  TCGA-AO-A12D  \n4  1.115180     NP_958781-# (plectin isoform 1e)  TCGA-AO-A12D  "
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "reshapedData[\"Gene\"]=reshapedData[\"RefSeq_accession_number\"].fillna('#')+\"-\"+reshapedData[\"gene_symbol\"].fillna('#')+\" (\"+reshapedData[\"gene_name\"].fillna('#')+\")\"\nreshapedData[\"Patient_ID\"]=\"TCGA-\" + reshapedData[\"Patient\"].str.slice(0,7)\nreshapedData.head()"
        }, 
        {
            "execution_count": 22, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "output_type": "stream", 
                    "name": "stdout", 
                    "text": "0\n"
                }
            ], 
            "source": "#verify what separator to use in out CSV to avoid further db load problem\ncoma_count=reshapedData['Gene'].str.contains(';').sum()\nprint (coma_count)"
        }, 
        {
            "execution_count": 23, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 23, 
                    "metadata": {}, 
                    "data": {
                        "text/plain": "{'asset_id': '25330465-2ad0-4e1b-8de1-fd154128a0f6',\n 'bucket_name': 'breastcancerproteomes-donotdelete-pr-zmcmbnmwpo1wza',\n 'file_name': 'Reshaped_data1.csv',\n 'message': 'File Reshaped_data1.csv has been written successfully to the associated OS'}"
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "project.save_data(data=reshapedData.to_csv(index=False, sep=';', encoding='utf-8'),file_name='Reshaped_data1.csv',overwrite=True)"
        }, 
        {
            "execution_count": 24, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 24, 
                    "metadata": {}, 
                    "data": {
                        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>RefSeq_accession_number</th>\n      <th>gene_symbol</th>\n      <th>gene_name</th>\n      <th>BH-A18Q.02TCGA</th>\n      <th>C8-A130.02TCGA</th>\n      <th>C8-A138.03TCGA</th>\n      <th>E2-A154.03TCGA</th>\n      <th>C8-A12L.04TCGA</th>\n      <th>A2-A0EX.04TCGA</th>\n      <th>AN-A04A.05TCGA</th>\n      <th>...</th>\n      <th>AR-A0TT.34TCGA</th>\n      <th>A2-A0SW.35TCGA</th>\n      <th>AO-A0JL.35TCGA</th>\n      <th>BH-A0BV.35TCGA</th>\n      <th>A2-A0YM.36TCGA</th>\n      <th>BH-A0C7.36TCGA</th>\n      <th>A2-A0SX.36TCGA</th>\n      <th>HEALTY1</th>\n      <th>HEALTY2</th>\n      <th>HEALTY3</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NP_958782</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1</td>\n      <td>0.195341</td>\n      <td>-0.494060</td>\n      <td>2.765081</td>\n      <td>0.862659</td>\n      <td>1.407570</td>\n      <td>1.185108</td>\n      <td>0.384588</td>\n      <td>...</td>\n      <td>-0.511421</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.065838</td>\n      <td>0.655850</td>\n      <td>-0.552212</td>\n      <td>-0.398560</td>\n      <td>0.598585</td>\n      <td>-0.191285</td>\n      <td>0.566975</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NP_958785</td>\n      <td>NaN</td>\n      <td>plectin isoform 1g</td>\n      <td>0.215413</td>\n      <td>-0.503899</td>\n      <td>2.779709</td>\n      <td>0.870186</td>\n      <td>1.407570</td>\n      <td>1.192612</td>\n      <td>0.371393</td>\n      <td>...</td>\n      <td>-0.526067</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.055893</td>\n      <td>0.658143</td>\n      <td>-0.547749</td>\n      <td>-0.392601</td>\n      <td>0.606697</td>\n      <td>-0.183918</td>\n      <td>0.578702</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NP_958786</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1a</td>\n      <td>0.215413</td>\n      <td>-0.500619</td>\n      <td>2.779709</td>\n      <td>0.870186</td>\n      <td>1.410312</td>\n      <td>1.188860</td>\n      <td>0.371393</td>\n      <td>...</td>\n      <td>-0.526067</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.065838</td>\n      <td>0.655850</td>\n      <td>-0.552212</td>\n      <td>-0.392601</td>\n      <td>0.603993</td>\n      <td>-0.186022</td>\n      <td>0.576747</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NP_000436</td>\n      <td>NaN</td>\n      <td>plectin isoform 1c</td>\n      <td>0.205377</td>\n      <td>-0.510459</td>\n      <td>2.797995</td>\n      <td>0.866423</td>\n      <td>1.407570</td>\n      <td>1.185108</td>\n      <td>0.377990</td>\n      <td>...</td>\n      <td>-0.533389</td>\n      <td>-0.487772</td>\n      <td>-0.10668</td>\n      <td>-0.055893</td>\n      <td>0.655850</td>\n      <td>-0.552212</td>\n      <td>-0.392601</td>\n      <td>0.603993</td>\n      <td>-0.186022</td>\n      <td>0.576747</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NP_958781</td>\n      <td>NaN</td>\n      <td>plectin isoform 1e</td>\n      <td>0.215413</td>\n      <td>-0.503899</td>\n      <td>2.787023</td>\n      <td>0.870186</td>\n      <td>1.413053</td>\n      <td>1.200116</td>\n      <td>0.374692</td>\n      <td>...</td>\n      <td>-0.529728</td>\n      <td>-0.503853</td>\n      <td>-0.10668</td>\n      <td>-0.062523</td>\n      <td>0.651264</td>\n      <td>-0.556675</td>\n      <td>-0.395581</td>\n      <td>0.603993</td>\n      <td>-0.167079</td>\n      <td>0.576747</td>\n    </tr>\n  </tbody>\n</table>\n<p>5 rows \u00d7 80 columns</p>\n</div>", 
                        "text/plain": "  RefSeq_accession_number gene_symbol           gene_name  BH-A18Q.02TCGA  \\\n0               NP_958782        PLEC   plectin isoform 1        0.195341   \n1               NP_958785         NaN  plectin isoform 1g        0.215413   \n2               NP_958786        PLEC  plectin isoform 1a        0.215413   \n3               NP_000436         NaN  plectin isoform 1c        0.205377   \n4               NP_958781         NaN  plectin isoform 1e        0.215413   \n\n   C8-A130.02TCGA  C8-A138.03TCGA  E2-A154.03TCGA  C8-A12L.04TCGA  \\\n0       -0.494060        2.765081        0.862659        1.407570   \n1       -0.503899        2.779709        0.870186        1.407570   \n2       -0.500619        2.779709        0.870186        1.410312   \n3       -0.510459        2.797995        0.866423        1.407570   \n4       -0.503899        2.787023        0.870186        1.413053   \n\n   A2-A0EX.04TCGA  AN-A04A.05TCGA    ...     AR-A0TT.34TCGA  A2-A0SW.35TCGA  \\\n0        1.185108        0.384588    ...          -0.511421       -0.487772   \n1        1.192612        0.371393    ...          -0.526067       -0.487772   \n2        1.188860        0.371393    ...          -0.526067       -0.487772   \n3        1.185108        0.377990    ...          -0.533389       -0.487772   \n4        1.200116        0.374692    ...          -0.529728       -0.503853   \n\n   AO-A0JL.35TCGA  BH-A0BV.35TCGA  A2-A0YM.36TCGA  BH-A0C7.36TCGA  \\\n0        -0.10668       -0.065838        0.655850       -0.552212   \n1        -0.10668       -0.055893        0.658143       -0.547749   \n2        -0.10668       -0.065838        0.655850       -0.552212   \n3        -0.10668       -0.055893        0.655850       -0.552212   \n4        -0.10668       -0.062523        0.651264       -0.556675   \n\n   A2-A0SX.36TCGA   HEALTY1   HEALTY2   HEALTY3  \n0       -0.398560  0.598585 -0.191285  0.566975  \n1       -0.392601  0.606697 -0.183918  0.578702  \n2       -0.392601  0.603993 -0.186022  0.576747  \n3       -0.392601  0.603993 -0.186022  0.576747  \n4       -0.395581  0.603993 -0.167079  0.576747  \n\n[5 rows x 80 columns]"
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "# I reimport cleaned CSV\n\nbody = client_8a4fc8dc8bf54832ab62f63af0c03772.get_object(Bucket='breastcancerproteomes-donotdelete-pr-zmcmbnmwpo1wza',Key='77_cancer_proteomes_CPTAC_itraq_NODouble.csv')['Body']\n# add missing __iter__ method, so pandas accepts body as file-like object\nif not hasattr(body, \"__iter__\"): body.__iter__ = types.MethodType( __iter__, body )\n\nnodoubled_data = pd.read_csv(body)\nnodoubled_data.head()"
        }, 
        {
            "execution_count": 25, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 25, 
                    "metadata": {}, 
                    "data": {
                        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>RefSeq_accession_number</th>\n      <th>gene_symbol</th>\n      <th>gene_name</th>\n      <th>Patient</th>\n      <th>Value</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NP_958782</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.195341</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NP_958785</td>\n      <td>NaN</td>\n      <td>plectin isoform 1g</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.215413</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NP_958786</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1a</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.215413</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NP_000436</td>\n      <td>NaN</td>\n      <td>plectin isoform 1c</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.205377</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NP_958781</td>\n      <td>NaN</td>\n      <td>plectin isoform 1e</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.215413</td>\n    </tr>\n  </tbody>\n</table>\n</div>", 
                        "text/plain": "  RefSeq_accession_number gene_symbol           gene_name         Patient  \\\n0               NP_958782        PLEC   plectin isoform 1  BH-A18Q.02TCGA   \n1               NP_958785         NaN  plectin isoform 1g  BH-A18Q.02TCGA   \n2               NP_958786        PLEC  plectin isoform 1a  BH-A18Q.02TCGA   \n3               NP_000436         NaN  plectin isoform 1c  BH-A18Q.02TCGA   \n4               NP_958781         NaN  plectin isoform 1e  BH-A18Q.02TCGA   \n\n      Value  \n0  0.195341  \n1  0.215413  \n2  0.215413  \n3  0.205377  \n4  0.215413  "
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "reshapedData1=pd.melt(nodoubled_data,id_vars=[\"RefSeq_accession_number\",\"gene_symbol\",\"gene_name\"],var_name=\"Patient\",value_name=\"Value\")\nreshapedData1.head()"
        }, 
        {
            "execution_count": 26, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 26, 
                    "metadata": {}, 
                    "data": {
                        "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>RefSeq_accession_number</th>\n      <th>gene_symbol</th>\n      <th>gene_name</th>\n      <th>Patient</th>\n      <th>Value</th>\n      <th>Gene</th>\n      <th>Patient_ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>NP_958782</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.195341</td>\n      <td>NP_958782-PLEC (plectin isoform 1)</td>\n      <td>TCGA-BH-A18Q</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>NP_958785</td>\n      <td>NaN</td>\n      <td>plectin isoform 1g</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.215413</td>\n      <td>NP_958785-# (plectin isoform 1g)</td>\n      <td>TCGA-BH-A18Q</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>NP_958786</td>\n      <td>PLEC</td>\n      <td>plectin isoform 1a</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.215413</td>\n      <td>NP_958786-PLEC (plectin isoform 1a)</td>\n      <td>TCGA-BH-A18Q</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>NP_000436</td>\n      <td>NaN</td>\n      <td>plectin isoform 1c</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.205377</td>\n      <td>NP_000436-# (plectin isoform 1c)</td>\n      <td>TCGA-BH-A18Q</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>NP_958781</td>\n      <td>NaN</td>\n      <td>plectin isoform 1e</td>\n      <td>BH-A18Q.02TCGA</td>\n      <td>0.215413</td>\n      <td>NP_958781-# (plectin isoform 1e)</td>\n      <td>TCGA-BH-A18Q</td>\n    </tr>\n  </tbody>\n</table>\n</div>", 
                        "text/plain": "  RefSeq_accession_number gene_symbol           gene_name         Patient  \\\n0               NP_958782        PLEC   plectin isoform 1  BH-A18Q.02TCGA   \n1               NP_958785         NaN  plectin isoform 1g  BH-A18Q.02TCGA   \n2               NP_958786        PLEC  plectin isoform 1a  BH-A18Q.02TCGA   \n3               NP_000436         NaN  plectin isoform 1c  BH-A18Q.02TCGA   \n4               NP_958781         NaN  plectin isoform 1e  BH-A18Q.02TCGA   \n\n      Value                                 Gene    Patient_ID  \n0  0.195341   NP_958782-PLEC (plectin isoform 1)  TCGA-BH-A18Q  \n1  0.215413     NP_958785-# (plectin isoform 1g)  TCGA-BH-A18Q  \n2  0.215413  NP_958786-PLEC (plectin isoform 1a)  TCGA-BH-A18Q  \n3  0.205377     NP_000436-# (plectin isoform 1c)  TCGA-BH-A18Q  \n4  0.215413     NP_958781-# (plectin isoform 1e)  TCGA-BH-A18Q  "
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "reshapedData1[\"Gene\"]=reshapedData1[\"RefSeq_accession_number\"].fillna('#')+\"-\"+reshapedData1[\"gene_symbol\"].fillna('#')+\" (\"+reshapedData1[\"gene_name\"].fillna('#')+\")\"\nreshapedData1[\"Patient_ID\"]=\"TCGA-\" + reshapedData1[\"Patient\"].str.slice(0,7)\nreshapedData1.head()"
        }, 
        {
            "execution_count": 27, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [
                {
                    "execution_count": 27, 
                    "metadata": {}, 
                    "data": {
                        "text/plain": "{'asset_id': '46644faf-3fe6-4479-9a86-f11cc7cdbefc',\n 'bucket_name': 'breastcancerproteomes-donotdelete-pr-zmcmbnmwpo1wza',\n 'file_name': 'Reshaped_data_nodouble.csv',\n 'message': 'File Reshaped_data_nodouble.csv has been written successfully to the associated OS'}"
                    }, 
                    "output_type": "execute_result"
                }
            ], 
            "source": "project.save_data(data=reshapedData1.to_csv(index=False, sep=';', encoding='utf-8'),file_name='Reshaped_data_nodouble.csv',overwrite=True)"
        }, 
        {
            "execution_count": null, 
            "cell_type": "code", 
            "metadata": {}, 
            "outputs": [], 
            "source": ""
        }
    ], 
    "metadata": {
        "kernelspec": {
            "display_name": "Python 3.5", 
            "name": "python3", 
            "language": "python"
        }, 
        "language_info": {
            "mimetype": "text/x-python", 
            "nbconvert_exporter": "python", 
            "version": "3.5.5", 
            "name": "python", 
            "file_extension": ".py", 
            "pygments_lexer": "ipython3", 
            "codemirror_mode": {
                "version": 3, 
                "name": "ipython"
            }
        }
    }, 
    "nbformat": 4
}